src/consumer.rs

```diff
@@ -37,6 +37,81 @@ impl<
     }
 }
 
+/// Continuously consumes message from a Queue.
+///
+/// A consumer represents a stream of messages created from
+/// the [basic.consume](https://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.consume) AMQP command.
+/// It continuously receives messages from the queue, as opposed to the
+/// [basic.get](https://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.get) command, which
+/// retrieves only a single message.
+///
+/// A consumer is obtained by calling [`Channel::basic_consume`] with the queue name.
+///
+/// New messages from this consumer can be accessed by obtaining the iterator from the consumer.
+/// This iterator returns new messages in the form of a
+/// [`Delivery`] for as long as the consumer is subscribed to the queue.
+///
+/// It is also possible to set a delegate to be spawned via [`set_delegate`].
+///
+/// ## Message acknowledgment
+///
+/// There are two ways for acknowledging a message:
+///
+/// * If the flag [`BasicConsumeOptions::no_ack`] is set to `true` while obtaining the consumer from
+///   [`Channel::basic_consume`], the server implicitely acknowledges each message after it has been
+///   sent.
+/// * If the flag [`BasicConsumeOptions::no_ack`] is set to `false`, a message has to be explicitely
+///   acknowledged or rejected with [`Channel::basic_ack`],
+///   [`Channel::basic_reject`] or [`Channel::basic_nack`]. See the documentation at [`Delivery`]
+///   for further information.
+///
+/// Also see the RabbitMQ documentation concerning
+/// [Consumer Acknowledgements and Publisher Confirms](https://www.rabbitmq.com/confirms.html).
```

Comment:
This is not related, it's the other way around. Publisher confirms are when you use channel.confirm_select, then the client waits for the server to send you some acks. It's not related to consumers at all

Reply:
I will link this document: https://www.rabbitmq.com/consumers.html#acknowledgement-modes